services/src/test/java/org/apache/druid/server/router/CoordinatorRuleManagerTest.java

```diff
@@ -109,7 +109,7 @@ public void testGetRulesWithKnownDatasourceReturningAllRulesWithDefaultRule()
     manager.poll();
     final List<Rule> rules = manager.getRulesWithDefault(DATASOURCE2);
     final List<Rule> expectedRules = new ArrayList<>();
-    expectedRules.add(new ForeverLoadRule(null));
+    expectedRules.add(new ForeverLoadRule(null, null));
```

Comment:
Maybe pass `true` explicitly in all tests where you want to retain the old behaviour. It would help you distinguish between tests for old and new behaviour.

Reply:
Would it make more sense to use null and false to differentiate old and new tests, since that is the real test for backward compatibility?

Reply:
Sure, that works too as long as there are at least a couple of tests for each of the 3 flavors: null (for backward compat), true and false.